apps/omg_watcher/test/omg_watcher/exit_processor/core_test.exs

```diff
@@ -1326,21 +1326,47 @@ defmodule OMG.Watcher.ExitProcessor.CoreTest do
   end
 
   describe "in-flight exit finalization" do
-    test "deactivate in-flight exit after all piggybacked outputs are finalized",
-         %{processor_empty: processor, transactions: [tx | _]} do
+    test "deactivate in-flight exit after all piggybacked transaction inputs are finalized",
```

Comment:
leave this test as it was and add a new test (or tests) that test your new behaviors

Reply:
another comment related to this. Judging by the test description, you're triggering exiting in `State` only when all piggybacks have finalized? I think this is too late - you should exit in `State` when a particular piggybacked in/output finalizes immediately.

If this is just about the test description, and you're exiting immediately, then the implementation is fine; splitting the test and/or amending the description will do!